data-prepper-plugins/otel-proto-common/src/main/java/org/opensearch/dataprepper/plugins/otel/codec/OTelProtoCodec.java

```diff
@@ -1244,22 +1244,29 @@ public static List<Bucket> createBuckets(List<Long> bucketCountsList, List<Doubl
             throw new IllegalArgumentException("OpenTelemetry protocol mandates that the number of elements in bucket_counts array must be by one greater than\n" +
                     "  // the number of elements in explicit_bounds array.");
         } else {
-            for (int i = 0; i < bucketCountsList.size(); i++) {
-                if (i == 0) {
-                    double min = -Float.MAX_VALUE; // "-Infinity"
-                    double max = explicitBoundsList.get(i);
-                    Long bucketCount = bucketCountsList.get(i);
-                    buckets.add(new DefaultBucket(min, max, bucketCount));
-                } else if (i == bucketCountsList.size() - 1) {
-                    double min = explicitBoundsList.get(i - 1);
-                    double max = Float.MAX_VALUE; // "Infinity"
-                    Long bucketCount = bucketCountsList.get(i);
-                    buckets.add(new DefaultBucket(min, max, bucketCount));
-                } else {
-                    double min = explicitBoundsList.get(i - 1);
-                    double max = explicitBoundsList.get(i);
-                    Long bucketCount = bucketCountsList.get(i);
-                    buckets.add(new DefaultBucket(min, max, bucketCount));
+            if (bucketCountsList.size() == 1) {
```

Comment:
Can you add a unit test which demonstrates the original failure and then shows that this fixes it?